ReactiveCocoaTests/Swift/FlattenSpec.swift

```diff
@@ -0,0 +1,200 @@
+				var disposed = false
+
+				beforeEach {
+					disposed = false
+					pipe.1.sendNext(SignalProducer<Int, TestError> { _, disposable in
+						disposable += ActionDisposable {
+							disposed = true
+						}
+					})
+				}
+
+				it("should dispose inner signals when outer signal interrupted") {
+					pipe.1.sendInterrupted()
+					expect(disposed).to(beTrue())
+				}
+
+				it("should dispose inner signals when outer signal failed") {
+					pipe.1.sendFailed(TestError.Default)
+					expect(disposed).to(beTrue())
+				}
+
+				it("should not dispose inner signals when outer signal completed") {
+					pipe.1.sendCompleted()
+					expect(disposed).to(beFalse())
+				}
+			}
+		}
+
+		describe("Signal.merge") {
+
+			var pipe: Pipe!
+			var disposable: Disposable?
+
+			beforeEach {
+				pipe = Signal.pipe()
+				disposable = pipe.0.flatten(.Merge).observe { _ in }
+			}
+
+			afterEach {
+				disposable?.dispose()
+			}
+
+			context("disposing") {
+				var disposed = false
+
+				beforeEach {
+					disposed = false
+					pipe.1.sendNext(SignalProducer<Int, TestError> { _, disposable in
+						disposable += ActionDisposable {
+							disposed = true
+						}
+					})
+				}
+
+				it("should dispose inner signals when outer signal interrupted") {
+					pipe.1.sendInterrupted()
+					expect(disposed).to(beTrue())
+				}
+
+				it("should dispose inner signals when outer signal failed") {
+					pipe.1.sendFailed(TestError.Default)
+					expect(disposed).to(beTrue())
+				}
+
+				it("should not dispose inner signals when outer signal completed") {
+					pipe.1.sendCompleted()
+					expect(disposed).to(beFalse())
+				}
+			}
+		}
+
+		describe("Signal.concat") {
+
+			var pipe: Pipe!
+			var disposable: Disposable?
+
+			beforeEach {
+				pipe = Signal.pipe()
+				disposable = pipe.0.flatten(.Concat).observe { _ in }
+			}
+
+			afterEach {
+				disposable?.dispose()
+			}
+
+			context("disposing") {
+				var disposed = false
+
+				beforeEach {
+					disposed = false
+					pipe.1.sendNext(SignalProducer<Int, TestError> { _, disposable in
+						disposable += ActionDisposable {
+							disposed = true
+						}
+					})
+				}
+
+				it("should dispose inner signals when outer signal interrupted") {
+					pipe.1.sendInterrupted()
+					expect(disposed).to(beTrue())
+				}
+
+				it("should dispose inner signals when outer signal failed") {
+					pipe.1.sendFailed(TestError.Default)
+					expect(disposed).to(beTrue())
+				}
+
+				it("should not dispose inner signals when outer signal completed") {
+					pipe.1.sendCompleted()
+					expect(disposed).to(beFalse())
+				}
+			}
+		}
+
+		describe("SignalProducer.switchToLatest") {
+			it("disposes original signal when result signal interrupted") {
+				
+				var disposed = false
+
+				let disposable = SignalProducer<SignalProducer<Void, NoError>, NoError> { observer, disposable in
+					disposable += ActionDisposable {
+						disposed = true
+					}
+				}.flatten(.Latest).start()
```

Comment:
Can you put each operator in its own line?
```
let disposable = SignalProducer<SignalProducer<(), NoError>, NoError> { observer, disposable in

}
  .flatten(.Latest)
  .start()
```

Note that we prefer to use `()` instead of `Void`.